### PR TITLE
Add test for issue #17231

### PR DESCRIPTION
**Pull request by Sporarum**

### CODE REVIEW

File: compiler/test-resources/repl/i17231 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+scala> summon[ValueOf["a"]]
+val res0: ValueOf["a"] = scala.ValueOf@61
```
Comment (nicolasstucki) [inline compiler/test-resources/repl/i17231:2]: This `toString` uses the hash. This test will not be stable.

Comment (Sporarum) [reply compiler/test-resources/repl/i17231:2]: Oh, I looked at the neighboring `i15493` which also had hashes, so I assumed it would be fine

Comment (Sporarum) [reply compiler/test-resources/repl/i17231:2]: How do you advise I proceed ?

Comment (dwijnand) [reply compiler/test-resources/repl/i17231:2]: The hash of ValueOf is of its underlying type, String, which is of the content.  "a" is hex 0x61, which isn't going to change.  So this is fine.

**Review by dwijnand — APPROVED** [review]

**Pull request merged**